### PR TITLE
NH-4005 - Dynamic component issue with keys sharing a same prefix

### PR DESCRIPTION
**Pull request by nfplee**
Test cases for https://nhibernate.jira.com/browse/NH-4005

### CODE REVIEW

**Comment by fredericDelaporte** [comment]
Thanks,

We may use directly this branch for fixing it by the way (with "Allow edits from maintainers.", if you let it enabled).

**Comment by nfplee** [comment]
No problem. Yeah I left that checkbox enabled. One minor thing I have noticed is the AddProperty method in the FixtureByCode.cs file has been marked as static when it doesn't need to be.

File: src/NHibernate.Test/NHSpecificTest/NH4005/FixtureByCode.cs [hunk 1]
```diff
@@ -58,25 +51,18 @@ private static void AddProperty(PersistentClass persistentClass, NHibernate.Mapp
...
 
-		protected override void CreateSchema()
-		{
-			base.CreateSchema();
-
-			ExecuteStatement("ALTER TABLE [Person] ADD [MainFullName] TEXT");
-			ExecuteStatement("ALTER TABLE [Person] ADD [MainEmail] TEXT");
-			ExecuteStatement("ALTER TABLE [Person] ADD [ContactFullName] TEXT");
-			ExecuteStatement("ALTER TABLE [Person] ADD [ContactEmail] TEXT");
-		}
```
Comment (hazzik) [inline src/NHibernate.Test/NHSpecificTest/NH4005/FixtureByCode.cs]: This is no more needed as we bond the column to a table correctly.

File: src/NHibernate.Test/NHSpecificTest/NH4005/FixtureByCode.cs [hunk 2]
```diff
@@ -58,25 +51,18 @@ private static void AddProperty(PersistentClass persistentClass, NHibernate.Mapp
...
 
-			simpleValue.AddColumn(new Column(name)
+			var column = new Column(name)
 			{
 				Value = simpleValue,
 				Length = 100,
 				IsNullable = true,
-			});
+			};
+			simpleValue.Table.AddColumn(column); // <--- HERE
```
Comment (hazzik) [inline src/NHibernate.Test/NHSpecificTest/NH4005/FixtureByCode.cs]: Here is a fix.

File: src/NHibernate.Test/NHSpecificTest/NH4005/FixtureByCode.cs [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+			// Add the "Name" property
+			var simpleValue = new SimpleValue(persistentClass.Table) { TypeName = "String" };
+
+			var column = new Column(name)
+			{
+				Value = simpleValue,
+				Length = 100,
+				IsNullable = true,
+			};
+			simpleValue.Table.AddColumn(column); // <--- HERE
```
Comment (hazzik) [inline src/NHibernate.Test/NHSpecificTest/NH4005/FixtureByCode.cs]: Here is the fix

File: src/NHibernate.Test/NHSpecificTest/NH4005/FixtureByCode.cs (hunk 1 above)
Comment (hazzik) [inline src/NHibernate.Test/NHSpecificTest/NH4005/FixtureByCode.cs]: No more needed as we correctly bond the column to it's table

**Pull request closed without merging**